roqoqo-qasm/src/backend.rs

```diff
@@ -33,6 +35,24 @@ use std::usize;
 /// computing platforms.
 ///
 ///
+fn process_operation_circuit<'a>(
+    circuit: impl Iterator<Item = &'a Operation>,
+    qasm_version: QasmVersion,
+    already_seen_declarations: &mut Vec<String>,
+    declarations: &mut String,
+) -> Result<(), RoqoqoBackendError> {
+    for operation in circuit {
+        if !already_seen_declarations.contains(&operation.hqslang().to_string()) {
+            already_seen_declarations.push(operation.hqslang().to_string());
+            declarations.push_str(&gate_definition(operation, qasm_version)?);
+            if !declarations.is_empty() {
+                declarations.push('\n');
+            }
+        }
+    }
+    Ok(())
+}
+
```

Comment:
I think this has to simply be moved elsewhere in the script, or the docstring won't be related to the Backend struct anymore.